052-record-reconciled-versions.md

```diff
@@ -0,0 +1,186 @@
+# Record Reconciled Version in Custom Resource
+
+For the `Kafka` Custom Resource it is often unclear when an upgrade has finished, whether all the brokers have rolled and to which version of the operator a cluster has been reconciled with.
+The following proposal puts forward a mechanism to make this clear in the custom resource, making it easy to programmatically check for upgrade completion, and could also be used/adopted in the upgrade system tests, simplifying the code and making for a slightly better upgrade UX story.
+
+## Current situation
+
+Currently the only way to know if an upgrade has finished to completion is to check is all the zookeeper and kafka brokers have rolled.
+https://github.com/strimzi/strimzi-kafka-operator/blob/2a1fdf9d8695bb22a2bf977b0ba5414291530207/systemtest/src/test/java/io/strimzi/systemtest/upgrade/KafkaUpgradeDowngradeST.java#L292
+Though this works, from a user standpoint it is far from ideal as it assumes a level of user knowledge about the cluster that may not be appropriate for all users.
+
+## Motivation
+
+The reason this proposal should be considered is it will allow a user to rely on the `Kafka` custom resource as their single source of truth, instead of watching/querying the individual kafka/zookeeper pods.
```

Comment:
```suggestion
This proposal allows a user to rely on the `Kafka` custom resource as their single source of truth, instead of watching/querying the individual kafka/zookeeper pods.
```